xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp

```diff
@@ -123,14 +128,44 @@ bool CVAAPIContext::EnsureContext(CVAAPIContext **ctx, CDecoder *decoder)
   return true;
 }
 
+namespace
+{
+int find_open_render_node_fd()
```

Comment:
I am saying that with the implementation of vaapi drm code, it was exactly (!) designed and implemented to add a drm node into the code. 

The only reason for depending on a server _was_ the rendering part. I emphasize: "was" - we currently depend on X11 without a single reason as this only provides us vaPutSurface which we don't use - we don't use any X11 specifics in our VAAPI. Depending on it is useless.

Then DRM and DMA sharing infrastructure was implemented afterwards - after wayland, long after X11 - and now vaapi can be used without any dependency to any display server, while rendering can be done with a sane API.

if you check the wayland backend, you will see that only vaPutSurface again depends on this display server dependency.